docs/architecture.md

```diff
@@ -0,0 +1,223 @@
+# Architecture
+
+The core of OSP-Magnum is best described as a library that provides a set of functions and data structures intended to support a space flight simulator. The user is responsible for implementing their own application on the top level, allowing for full control over every feature.
+
+OSP-Magnum provides two distinct worlds the application can manage:
+* **Universe**: A space simulator that can handle massive world sizes with multiple coordinate spaces. Objects in space are referred to as **Satellites**.
+* **ActiveScene**: A typical game engine that can support physics, rendering, etc..
+
+The ActiveScene can be synchronized to represent a small area of the Universe, where in-game representations of nearby Satellites can be added to the scene. This area is referred to as the **ActiveArea** and can follow the user around to seamlessly explore the Universe while also avoiding floating point precision errors.
```

Comment:
I'd point out that the engine doesn't inherently limit anything to a single ActiveArea.

We don't have multiple set up, but there's no reason why we couldn't if we wanted to.